### ISSUE TITLE
Code

### ISSUE THREAD

**Issue by bhagwatsolunke** [bhagwatsolunke/dl#1]
Pract 1
*************
import tensorflow as tf
print(tf.__version__)
print(tf.reduce_sum(tf.random.normal([1000,1000])))

from tensorflow import keras
#keras is a high level api fro tensorflow
from keras import datasets
#importing mnist dataset of handwrittend digits
import pandas as pd
#load MNIST data
from keras.datasets import mnist
# mnist.load_data() is function that return two tuples with two arrays each
(train_images,train_labes),(test_images,test_labels) = mnist.load_data()
train_images.shape,test_images.shape

#Tensors are multidimensional arrays used as containers to store data
#Tensors are fundatmental way of storing data in ML, they have rank giving an idea of there dimensions
# if 1D they one vector and 2D for matrixes , they have shapes dtypes such as float64 or strings

#Here we are creating a tensor constant x with val 5.0 and datatype float64 for double precision
x = tf.constant(5.0,dtype=tf.float64)
y = tf.constant(7.0,dtype=tf.float64)
# we perform a addition of tensors which creates a computation graph which represent addition
z = x+y
#z.numpy() function extracts the values of tensors
result = z.numpy()
print(result)

import torch
x = torch.tensor(5.0,dtype=torch.float64)
y = torch.tensor(7.0,dtype=torch.float64)
z = x+y
result = z.item()
print(result)

*********
Pract 2
*********
import tensorflow as tf
from tensorflow import keras
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

mnist = tf.keras.datasets.mnist
#x_train as images y_train is labes
(x_train,y_train),(x_test,y_test) = mnist.load_data()

len(x_train)
len(x_test)
len(x_test)
len(y_train)
x_train.shape
x_train[0]

#matshow is function used to 2D matrices as images
plt.matshow(x_train[18888])

plt.imshow(-x_train[0],cmap="gray")

#Normalize the images by scaling pixel instensties to the range 0,1
#Divide by 255 as the pixel vary from 0 black to 255 white
x_train = x_train / 255
x_test = x_test / 255
x_train[0]

#keras api has Sequential model which allows to create each layer
#Flatten layer is the first layer that reshapes ,its takes a 28 by 28 and flattens it in 1D 784 elements array
#Dense layer is the output layer with 10 classes for 10 digits
model = keras.Sequential([
    keras.layers.Flatten(input_shape=(28,28)),
    keras.layers.Dense(128,activation="relu"),
    keras.layers.Dense(10,activation="softmax")
])
model.summary()

#optimizer defines the algorithm used for training sgd stand for Stochastic Gradient Descent
#metrics is used to evaluates a models perfomance

model.compile(optimizer = "sgd",
              loss="sparse_categorical_crossentropy",
              metrics=['accuracy'])

#evaluating the model
test_loss , test_acc = model.evaluate(x_test,y_test)
print("Loss=%.3f" %test_loss)
print("Accuracy=%.3f" %test_acc)

import random
n = random.randint(0,9999)
plt.imshow(x_test[n])
plt.show()

predicted_value = model.predict(x_test)
print("Handwritten number in the image is =%d" %np.argmax(predicted_value[0]))

history.history.keys()

plt.plot(history.history['accuracy'])
plt.plot(history.history['val_accuracy'])
plt.title('model accuracy')
plt.ylabel('accuracy')
plt.xlabel('epoch')
plt.legend(['Train','Validation'],loc ='upper left')
plt.show()

history = model.fit(x_train,y_train,validation_data=(x_test,y_test),epochs=10)

************
Pract 3
************
